diff --git a/CMakeLists.txt b/CMakeLists.txt
index a045c60..e22f3c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,9 @@ if(NOT OpenVINODeveloperPackage_FOUND)
     find_package(OpenVINO REQUIRED COMPONENTS Runtime OPTIONAL_COMPONENTS TensorFlow)
 endif()

+# wa for conda channel in 2024.1 release
+set(OpenVINO_Frontend_TensorFlow_FOUND OFF)
+
 # set the project version using OpenVINO major, minor and patch versions
 set(CMAKE_PROJECT_VERSION_TWEAK 0)
 set(CMAKE_PROJECT_VERSION "${OpenVINO_VERSION}.${CMAKE_PROJECT_VERSION_TWEAK}")
