diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2ca8e07..207dab0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -68,7 +68,8 @@ FetchContent_GetProperties(sentencepiece)
 if(NOT sentencepiece_POPULATED)
   set(SPM_ENABLE_SHARED OFF CACHE BOOL "")
   set(SPM_ENABLE_TCMALLOC OFF CACHE BOOL "")
-  set(SPM_USE_BUILTIN_PROTOBUF ON)
+  set(SPM_PROTOBUF_PROVIDER "package" CACHE STRING "")
+  set(SPM_ABSL_PROVIDER "package" CACHE STRING "")
   FetchContent_Populate(sentencepiece)
   add_subdirectory(${sentencepiece_SOURCE_DIR} ${sentencepiece_BINARY_DIR} EXCLUDE_FROM_ALL)
 endif()
